src/pyhf/workspace.py

```diff
@@ -743,3 +743,27 @@ def sorted(cls, workspace):
         newspec['observations'].sort(key=lambda e: e['name'])
 
         return cls(newspec)
+
+
+def build(model, data, name='measurement'):
+    """
+    Build a workspace from model and data.
+
+    Args:
+        model (~pyhf.pdf.Model): A model to store into a workspace
+        model (tensor): A array holding observations to store into a workspace
+
+    Returns:
+        ~pyhf.workspace.Workspace: A new  workspace object
+
+    """
+    workspace = copy.deepcopy(model.spec)
```

Comment:
i think we should be a bit explicit here, no?
```suggestion
    workspace = dict(channels=copy.deepcopy(model.spec['channels']))
```